File: frontend/app/models/send-alert-config.js
import DS from 'ember-data';

export default DS.Model.extend({
    alert: DS.belongsTo('alert'),
    message_template: DS.attr('string'),
    comm_type: DS.attr('string'), 
    to_addresses: DS.attr(),
    subject_template: DS.attr('string'),
    inserted_at: DS.attr('utc'),
    updated_at: DS.attr('utc'),
});

File: web/models/send_alert_config.ex
defmodule AfterGlow.SendAlertConfig do
  use AfterGlow.Web, :model
  alias AfterGlow.Alert
  import EctoEnum, only: [defenum: 2]
  
  defenum CommTypeEnum, email: 0, sms: 1, call: 2
  schema "generated_alerts" do
    belongs_to :alert, Alert
    field :message_template, :string
    field :comm_type, :integer
    field :to_addresses, {:array, :string}
    field :subject_template, :string
    timestamps()
  end

  @doc """
  Builds a changeset based on the `struct` and `params`.
  """
  def changeset(struct, params \\ %{}) do
    struct
    |> cast(params, [:alert_id, :message_template, :subject_template, :to_addresses, :comm_type])
    |> validate_required( [:alert_id, :message_template, :subject_template, :to_addresses, :comm_type])
  end
  
end
